File: src/main/scala/imcode/server/document/index/service/impl/UnavailableDocumentIndexService.scala
package imcode.server.document.index.service.impl

import _root_.imcode.server.document.index.service.{DocumentIndexService, IndexRebuildTask, IndexUpdateOp}
import com.imcode.imcms.api.ServiceUnavailableException
import org.apache.solr.client.solrj.SolrQuery
import org.apache.solr.client.solrj.response.QueryResponse

object UnavailableDocumentIndexService extends DocumentIndexService {

  override def query(solrQuery: SolrQuery): QueryResponse = throw new ServiceUnavailableException()

  override def update(request: IndexUpdateOp) {}

  override def rebuild(): IndexRebuildTask = throw new ServiceUnavailableException()

  override def currentRebuildTaskOpt(): Option[IndexRebuildTask] = None

  override def shutdown() {}
}

File: src/main/java/imcode/server/document/index/service/impl/UnavailableDocumentIndexService.java
package imcode.server.document.index.service.impl;

import com.imcode.imcms.api.exception.ServiceUnavailableException;
import imcode.server.document.index.service.DocumentIndexService;
import imcode.server.document.index.service.IndexUpdateOp;
import org.apache.solr.client.solrj.SolrQuery;
import org.apache.solr.client.solrj.response.QueryResponse;

import java.util.concurrent.Future;

// translated from scala
public class UnavailableDocumentIndexService implements DocumentIndexService {
    @Override
    public QueryResponse query(SolrQuery solrQuery) {
        throw new ServiceUnavailableException();
    }

    @Override
    public void update(IndexUpdateOp request) {
    }

    @Override
    public Future rebuild() {
        throw new ServiceUnavailableException();
    }

    @Override
    public void shutdown() {
    }

    @Override
    public boolean isUpdateDone() {
        throw new ServiceUnavailableException();
    }
}
